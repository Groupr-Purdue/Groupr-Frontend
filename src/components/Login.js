// @flow
import React from 'react';
import { RaisedButton, TextField } from 'material-ui';
import { Row, Col } from 'react-flexbox-grid-aphrodite';
import login from '~/store/loginForm';
import userStore from '~/store/user';
import router from '~/store/router';
import navbar from '~/store/navbar';

export const handleSubmit = (): void =>
  login.submit().then((user: Object): Object => {
    console.log(user); // eslint-disable-line no-console
    const currentUser = {
      firstName: user.first_name,
      lastName: user.last_name,
      careerAccount: user.career_account,
      token: user.token,
      loggedIn: true,
    }
    userStore.storeUser(currentUser);
    router.push(`/courses/${user.id}`);
  });

export default class Login extends React.Component {
  render(): Element {
    return (
      <Row >
        <Col xs={12} >
          <TextField
            floatingLabelText='Purdue Career Account'
            onChange={(ev: Object, value: string): void => login.careerAccount = value} />
        </Col>
        <Col xs={12} >
          <TextField
            floatingLabelText='Password'
            type='password'
            onChange={(ev: Object, value: string): void => login.password = value} />
        </Col>
        <Col xs={12}>
          <RaisedButton
            primary={true}
            onClick={handleSubmit} >
            Sign-In
            </RaisedButton>
        </Col>
      </Row>
    );
  }
}
