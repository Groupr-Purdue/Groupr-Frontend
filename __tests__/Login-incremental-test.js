import React from 'react';
import Login from '~/components/Login';
import { render } from 'enzyme';

test('jest Login incremental snapshot test', () => {
  expect(<Login />).toMatchSnapshot();
});


// TODO Test loginUser from  user store