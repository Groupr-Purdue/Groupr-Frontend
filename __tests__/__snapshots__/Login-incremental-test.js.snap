exports[`test jest Login incremental snapshot test 1`] = `<Login />`;
